.github/workflows/ci.yml

```diff
@@ -197,3 +197,31 @@ jobs:
         with:
           name: Test Results (${{ matrix.version }})
           path: junit.xml
+
+  coverge-test:
+    name: Generate code coverage
+    runs-on: ubuntu-latest
+    needs: build-and-lint
+    timeout-minutes: 10   
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Run Tests
+        run: ./run-tests.sh 6.1
```

Comment:
This doesn't make much sense. `vm-test` has a test matrix that runs this against a wide range of kernel versions, and we need test coverage for each of them.

The point is to add coverage reporting to the existing `vm-test`, `test-on-prev-go` and `test-on-arm64` jobs. Ideally, this would all be done in a single step to keep the size of the yaml a bit manageable:

```yaml
      - run: go tool covdata merge -o coverage-merged.out
      - run: go tool cover -html=coverage-merged.out -o coverage.html

      - name: Upload Coverage Report
        uses: actions/upload-artifact@v3
        with:
          name: coverage-report
          path: |
            ./coverage-merged.out
            ./coverage.html
```

Reply:
Will I have to add this to all three job or it can be done in another separate job?

Reply:
If we want coverage for all three jobs (we do), then yes, it needs to be in all three. :slightly_smiling_face: